tests/lib/core/e-cmd-pattern-control-source.c

```diff
@@ -41,9 +41,16 @@ test_setup (void)
   song = bt_song_new (app);
   bt_child_proxy_get ((gpointer) song, "sequence", &sequence, "song-info",
       &song_info, "song-info::tick-duration", &tick_time, NULL);
-  src1 = BT_MACHINE (bt_source_machine_new (song, "gen1",
+
+  BtMachineConstructorParams cparams;
+  cparams.id = "gen1";
+  cparams.song = song;
```

Comment:
Maybe we can add a macro for this (in a follow-up change) to make this a one-liner.